sdk/java/pulumi/src/test/java/com/pulumi/deployment/DeploymentExceptionTest.java

```diff
@@ -40,14 +41,14 @@ static void cleanup() {
     void testUrnFutureDoesNotHangOnException() {
         mock.standardLogger.setLevel(Level.OFF);
 
-        assertThatThrownBy(() -> mock.testAsync(MyIncorrectStack::new).join())
+        assertThatThrownBy(() -> mock.testAsync(MyIncorrectStack::init).join())
                 .getRootCause()
                 .isInstanceOf(RunException.class)
                 .hasMessageContaining(DeliberateException.class.getName());
     }
 
-    public static class MyIncorrectStack extends Stack {
-        public MyIncorrectStack() {
+    public static class MyIncorrectStack {
```

Comment:
no need for a class, a method is enough `public static void myIncorrectStack(Context ctx)`